pkgs/applications/misc/warpd/default.nix

```
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, git
+, libXi
+, libXinerama
+, libXft
+, libXfixes
+, libXtst
+, libX11
+, libXext
+, waylandSupport ? false, cairo, libxkbcommon, wayland
+}:
+
+stdenv.mkDerivation rec {
+  pname = "warpd";
+  version = "1.3.2";
+
+  src = fetchFromGitHub {
+    owner = "rvaiya";
+    repo = "warpd";
+    rev = "v${version}";
+    sha256 = "AR/uLgNX1VLPEcfUd8cnplMiaoEJlUxQ55Fst62RnbI=";
+    leaveDotGit = true;
+  };
+
+  nativeBuildInputs = [ git ];
+
+  buildInputs =  [
+    libXi
+    libXinerama
+    libXft
+    libXfixes
+    libXtst
+    libX11
+    libXext
+  ] ++ lib.optionals waylandSupport [
+    cairo
+    libxkbcommon
+    wayland
+  ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace 'install -m644 warpd.1.gz $(DESTDIR)$(PREFIX)/share/man/man1' 'install -D -m644 warpd.1.gz -t $(DESTDIR)$(PREFIX)/share/man/man1'
+
+    substituteInPlace Makefile \
+      --replace 'install -m755 bin/warpd $(DESTDIR)$(PREFIX)/bin/' 'install -D -m755 bin/warpd -t $(DESTDIR)$(PREFIX)/bin/'
+  '';
```

Something like this would've been sufficient :^)

```suggestion
  postPatch = ''
    substituteInPlace Makefile \
      --replace '-m644' '-Dm644' \
      --replace '-m755' '-Dm755'
  '';
```

---

This by itself won't build---investigating `install`'s man page, -D doesn't build the last directory unless `-t` is passed too:

       -D     create all leading components of DEST except the last, or
              all components of --target-directory, then copy SOURCE to
              DEST

so I added a bit in the commit to replace the install paths with a `-t` prefix, being careful not to replace them throughout every occurrence in the Makefile :)